<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Mohammad Al-Turany/Sergey Linev</author>
 <class>TGo4ConfigStep</class>
 <widget class="QWidget" name="TGo4ConfigStep">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>436</width>
    <height>282</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Configuration </string>
  </property>
  <layout class="QGridLayout">
   <property name="margin">
    <number>2</number>
   </property>
   <property name="spacing">
    <number>0</number>
   </property>
   <item row="0" column="0">
    <widget class="QGroupBox" name="ControlGroup_2">
     <property name="sizePolicy">
      <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Step Control</string>
     </property>
     <layout class="QHBoxLayout">
      <property name="spacing">
       <number>1</number>
      </property>
      <property name="margin">
       <number>3</number>
      </property>
      <item>
       <widget class="QCheckBox" name="EnableStepBox">
        <property name="toolTip">
         <string>Enable processing of this step.</string>
        </property>
        <property name="text">
         <string>Enable Step</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="EnableSourceBox">
        <property name="toolTip">
         <string>Enable event source. If disabled, step will then process output events of previous steps.</string>
        </property>
        <property name="text">
         <string>Source</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="EnableStoreBox">
        <property name="toolTip">
         <string>Enable storing output events of this step.</string>
        </property>
        <property name="text">
         <string>Store</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QGroupBox" name="StoreBox">
     <property name="sizePolicy">
      <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Event store</string>
     </property>
     <layout class="QGridLayout">
      <property name="margin">
       <number>3</number>
      </property>
      <property name="spacing">
       <number>1</number>
      </property>
      <item row="0" column="0">
       <widget class="QComboBox" name="OutputCombo">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Type of event store.</string>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <item>
         <property name="text">
          <string>Go4FileStore (1 tree/step) (*.root)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Go4BackStore</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="0">
       <layout class="QHBoxLayout">
        <item>
         <widget class="QLabel" name="InputSource_1_2_2">
          <property name="text">
           <string>Name:</string>
          </property>
          <property name="wordWrap">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="StoreNameEdit">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>Event store name.</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="FileNameOutput">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>30</width>
            <height>25</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>30</width>
            <height>25</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Browse for output file.</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset>:/icons/findfile.png</iconset>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="0">
       <layout class="QHBoxLayout">
        <item>
         <widget class="QSpinBox" name="SplitLevel">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>100</width>
            <height>32767</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Root TTree split level.</string>
          </property>
          <property name="maximum">
           <number>99</number>
          </property>
          <property name="value">
           <number>99</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="BufferSize">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>70</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>50</width>
            <height>32767</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Root branch buffer size.</string>
          </property>
          <property name="suffix">
           <string> kB  </string>
          </property>
          <property name="minimum">
           <number>4</number>
          </property>
          <property name="maximum">
           <number>128</number>
          </property>
          <property name="value">
           <number>4</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="CompLevel">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>50</width>
            <height>32767</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Root TFile compression level.</string>
          </property>
          <property name="maximum">
           <number>9</number>
          </property>
          <property name="value">
           <number>5</number>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="spacer5">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Expanding</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>99</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QCheckBox" name="StoreOverwriteMode">
          <property name="maximumSize">
           <size>
            <width>32767</width>
            <height>24</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Overwrite old file, or append.</string>
          </property>
          <property name="text">
           <string>&amp;Overwrite</string>
          </property>
          <property name="shortcut">
           <string>Alt+O</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QGroupBox" name="SourceBox">
     <property name="sizePolicy">
      <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="font">
      <font/>
     </property>
     <property name="title">
      <string>Event source</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="spacing">
       <number>1</number>
      </property>
      <property name="margin">
       <number>3</number>
      </property>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QComboBox" name="EventSourceCombo">
          <property name="toolTip">
           <string>Type of event source.</string>
          </property>
          <item>
           <property name="text">
            <string>Go4FileSource (1 tree/step) (*.root)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>MBS File (*.lmd *.LMD *.lml)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>MBS Stream Server</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>MBS Transport Server</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>MBS Event Server</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Remote Event Server</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>MBS Random</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>User Source</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="MbsMonitorBtn">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>30</width>
            <height>25</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>30</width>
            <height>25</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Open MBS monitor.</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset>:/icons/mbslogo.png</iconset>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout">
        <item>
         <widget class="QLabel" name="InputSource_1_3">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Name:</string>
          </property>
          <property name="wordWrap">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="SourceNameEdit">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>File name or node name.</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="FileNameInput">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>30</width>
            <height>25</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>30</width>
            <height>25</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Browse for input file.</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset>:/icons/findfile.png</iconset>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <layout class="QHBoxLayout">
          <property name="spacing">
           <number>6</number>
          </property>
          <property name="margin">
           <number>0</number>
          </property>
          <item>
           <widget class="QLabel" name="TextLabelTagfile">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Tagfile:</string>
            </property>
            <property name="wordWrap">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="LineEditTagfile">
            <property name="sizePolicy">
             <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
              <horstretch>1</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Optional MBS tag file name.</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLabel" name="TextLabelPortNumber">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>Port:</string>
            </property>
            <property name="wordWrap">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="SpinBoxPortNumber">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>1</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="specialValueText">
             <string>default</string>
            </property>
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>99999</number>
            </property>
            <property name="toolTip">
             <string>Port number, 0 - default (depends from the source kind)</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout17">
          <item>
           <widget class="QLabel" name="TextLabelRetryNumber">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>Retr:</string>
            </property>
            <property name="wordWrap">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="SpinBoxRetryNumber">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>1</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="specialValueText">
             <string>never</string>
            </property>
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>9999</number>
            </property>
            <property name="toolTip">
             <string>How many times source will rerty to connect with server, 0 - never</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        
        <item>
         <layout class="QHBoxLayout">
          <property name="spacing">
           <number>6</number>
          </property>
          <property name="margin">
           <number>0</number>
          </property>
          <item>
           <widget class="QLabel" name="TextLabelArgs">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string> Args:</string>
            </property>
            <property name="wordWrap">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="LineEditArgs">
            <property name="sizePolicy">
             <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
              <horstretch>1</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Optional user arguments.</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout">
        <item>
         <widget class="QSpinBox" name="SpinBoxStartEvent">
          <property name="toolTip">
           <string>Sequence number of first event for processing</string>
          </property>
          <property name="maximum">
           <number>2000000000</number>
          </property>
          <property name="singleStep">
           <number>1000</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="SpinBoxStopEvent">
          <property name="toolTip">
           <string>Last event to process (default - all events are processed)</string>
          </property>
          <property name="specialValueText">
           <string>all</string>
          </property>
          <property name="maximum">
           <number>2000000000</number>
          </property>
          <property name="singleStep">
           <number>1000</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="SpinBoxInterEvent">
          <property name="toolTip">
           <string>Skipping interval between two events to be processed.</string>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>999999999</number>
          </property>
          <property name="singleStep">
           <number>1000</number>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="spacer6">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Expanding</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>5</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QSpinBox" name="SpinBoxTimeout">
          <property name="toolTip">
           <string>Socket timeout for MBS sources.</string>
          </property>
          <property name="specialValueText">
           <string>none</string>
          </property>
          <property name="suffix">
           <string> s</string>
          </property>
          <property name="maximum">
           <number>9999</number>
          </property>
          <property name="value">
           <number>1</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>EnableStepBox</sender>
   <signal>stateChanged(int)</signal>
   <receiver>TGo4ConfigStep</receiver>
   <slot>StepStateChanged(int)</slot>
  </connection>
  <connection>
   <sender>EnableSourceBox</sender>
   <signal>stateChanged(int)</signal>
   <receiver>TGo4ConfigStep</receiver>
   <slot>InputStateChanged(int)</slot>
  </connection>
  <connection>
   <sender>EnableStoreBox</sender>
   <signal>stateChanged(int)</signal>
   <receiver>TGo4ConfigStep</receiver>
   <slot>OutputStateChanged(int)</slot>
  </connection>
  <connection>
   <sender>SourceNameEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>TGo4ConfigStep</receiver>
   <slot>InputSourceText(QString)</slot>
  </connection>
  <connection>
   <sender>EventSourceCombo</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>TGo4ConfigStep</receiver>
   <slot>SourceComboHighlighted(int)</slot>
  </connection>
  <connection>
   <sender>OutputCombo</sender>
   <signal>activated(int)</signal>
   <receiver>TGo4ConfigStep</receiver>
   <slot>StoreComboHighlighted(int)</slot>
  </connection>
  <connection>
   <sender>StoreNameEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>TGo4ConfigStep</receiver>
   <slot>OutputNameText(QString)</slot>
  </connection>
  <connection>
   <sender>FileNameOutput</sender>
   <signal>clicked()</signal>
   <receiver>TGo4ConfigStep</receiver>
   <slot>OutputFileDialog()</slot>
  </connection>
  <connection>
   <sender>CompLevel</sender>
   <signal>valueChanged(int)</signal>
   <receiver>TGo4ConfigStep</receiver>
   <slot>StoreCompLevel(int)</slot>
  </connection>
  <connection>
   <sender>BufferSize</sender>
   <signal>valueChanged(int)</signal>
   <receiver>TGo4ConfigStep</receiver>
   <slot>StoreBufferSize(int)</slot>
  </connection>
  <connection>
   <sender>SplitLevel</sender>
   <signal>valueChanged(int)</signal>
   <receiver>TGo4ConfigStep</receiver>
   <slot>StoreSplitLevel(int)</slot>
  </connection>
  <connection>
   <sender>StoreOverwriteMode</sender>
   <signal>toggled(bool)</signal>
   <receiver>TGo4ConfigStep</receiver>
   <slot>StoreOverWrite(bool)</slot>
  </connection>
  <connection>
   <sender>LineEditArgs</sender>
   <signal>textChanged(QString)</signal>
   <receiver>TGo4ConfigStep</receiver>
   <slot>InputArguments(QString)</slot>
  </connection>
  <connection>
   <sender>LineEditTagfile</sender>
   <signal>textChanged(QString)</signal>
   <receiver>TGo4ConfigStep</receiver>
   <slot>InputTagfile(QString)</slot>
  </connection>
  <connection>
   <sender>SpinBoxStartEvent</sender>
   <signal>valueChanged(int)</signal>
   <receiver>TGo4ConfigStep</receiver>
   <slot>ChangeStartEvent(int)</slot>
  </connection>
  <connection>
   <sender>SpinBoxStopEvent</sender>
   <signal>valueChanged(int)</signal>
   <receiver>TGo4ConfigStep</receiver>
   <slot>ChangeStopEvent(int)</slot>
  </connection>
  <connection>
   <sender>FileNameInput</sender>
   <signal>clicked()</signal>
   <receiver>TGo4ConfigStep</receiver>
   <slot>InputFileDialog()</slot>
  </connection>
  <connection>
   <sender>SpinBoxInterEvent</sender>
   <signal>valueChanged(int)</signal>
   <receiver>TGo4ConfigStep</receiver>
   <slot>ChangeEventInterval(int)</slot>
  </connection>
  <connection>
   <sender>SpinBoxTimeout</sender>
   <signal>valueChanged(int)</signal>
   <receiver>TGo4ConfigStep</receiver>
   <slot>ChangeSourceTimeout(int)</slot>
  </connection>
  <connection>
   <sender>MbsMonitorBtn</sender>
   <signal>clicked()</signal>
   <receiver>TGo4ConfigStep</receiver>
   <slot>MbsMonitorBtn_clicked()</slot>
  </connection>
  <connection>
   <sender>SpinBoxPortNumber</sender>
   <signal>valueChanged(int)</signal>
   <receiver>TGo4ConfigStep</receiver>
   <slot>InputPortChanged(int)</slot>
  </connection>
  <connection>
   <sender>SpinBoxRetryNumber</sender>
   <signal>valueChanged(int)</signal>
   <receiver>TGo4ConfigStep</receiver>
   <slot>RetryCntChanged(int)</slot>
  </connection>
 </connections>
</ui>
